tests/test.watcher.js

```diff
@@ -93,14 +93,23 @@ describe('watcher', () => {
         onChange: sinon.spy(() => {}),
       };
 
-      proxyFileChanges({...conf, filePath: '/somewhere/.git'});
+      proxyFileChanges({...conf, filePath: '/somewhere/freaky'});
       assert.equal(conf.onChange.called, false);
 
       proxyFileChanges({...conf, filePath: '/any/file/'});
       assert.equal(conf.onChange.called, true);
 
     });
 
+    it('filters out commonly unwanted files by default', () => {
+      const conf = {
+        ...defaults, shouldWatchFile: undefined,
+        onChange: sinon.spy(() => {}),
+      };
+      proxyFileChanges({...conf, filePath: '/somewhere/.git'});
```

Comment:
I don't think it's necessary to re-test all of the logic already tested by the file filter tests, such as [ignoring XPIs](https://github.com/mozilla/web-ext/blob/master/tests/test-cmd/test.build.js#L181). Ideally, I wanted to do something more like `assert.isInstance(filter, build.FileFilter)` but I couldn't think of an easy way to do it.